src/main/java/fr/cnes/sonarqube/plugins/framac/report/FramaCReportReader.java

```diff
@@ -211,4 +178,48 @@ private void valueRulesMatchingPatterns() {
 		return errors;
 	}
 
+	public List<FramaCError> parseOut(Path fileReportPath) {
+		LOGGER.info("Parsing out report: " + fileReportPath.getFileName() + " (Beginning)");
+
+		List<FramaCError> errors = new ArrayList<>();
+
+		try (Scanner scanner = new Scanner(fileReportPath, ENCODING.name())) {
+			while (scanner.hasNextLine()) {
+				String line = scanner.nextLine();
+				FramaCError error = getKernelError(line);
+				if (error != null) {
+					if(error.getDescription() == null
+						|| error.getDescription().length() == 0) {
+						// Get the descritpion on the next line
+						if(scanner.hasNextLine()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonar-frama-c-plugin&pullRequest=23&issues=AWY0CuQ5vOWp7Yd6eg9n&open=AWY0CuQ5vOWp7Yd6eg9n)

[ISSUE_KEY]: AWY0CuQ5vOWp7Yd6eg9n